controllers/idle_handlers.go

```diff
@@ -18,16 +18,55 @@ package controllers
 
 import (
 	"context"
+	"fmt"
+	"os"
 
 	ranv1alpha1 "github.com/openshift-kni/lifecycle-agent/api/v1alpha1"
+	"github.com/openshift-kni/lifecycle-agent/controllers/utils"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
+func getOSName(seedImageVersion string) string {
+	return fmt.Sprintf("rhcos_%s", seedImageVersion)
+}
+
+func (r *ImageBasedUpgradeReconciler) removeDeployment(osName string) {
+	status, err := r.RpmOSTreeClient.QueryStatus()
+	if err != nil {
+		r.Log.Error(err, "Failed to query rpm-ostree")
+		return
+	}
+	for index, deployment := range status.Deployments {
+		if deployment.OSName == osName {
+			ostreeUndeployCommand := fmt.Sprintf("ostree admin undeploy %d", index)
+			if err := utils.ExecuteChrootCmd(utils.Host, ostreeUndeployCommand); err != nil {
+				r.Log.Error(err, "Remove deployment failed", "osName", osName, "index", index)
+			}
+			return
+		}
+	}
+	r.Log.Info("Unable to find ostree deployment", "osName", osName)
+}
+
 //nolint:unparam
 func (r *ImageBasedUpgradeReconciler) handleAbort(ctx context.Context, ibu *ranv1alpha1.ImageBasedUpgrade) (ctrl.Result, error) {
+	OSName := getOSName(ibu.Spec.SeedImageRef.Version)
+	r.removeDeployment(OSName)
 
-	// TODO actual steps
-	// If succeeds, return doNotRequeue
+	sysrootDeployPath := fmt.Sprintf("/sysroot/ostree/deploy/%s", OSName)
+	if _, err := os.Stat(sysrootDeployPath); err != nil {
+		removeSysrootFilesCommand := fmt.Sprintf("unshare -m /bin/sh -c \"mount -o remount,rw /sysroot && rm -rf %s\"", sysrootDeployPath)
+		if err := utils.ExecuteChrootCmd(utils.Host, removeSysrootFilesCommand); err != nil {
+			r.Log.Error(err, "Failed to remove sysroot files")
+		}
+	}
```

Comment:
if the previous command fails, does this leave the node in a bad state?

Reply:
@jc-rh yes, very possibly. And if we proceeded to wiping the stateroot, it would be in a worse state. We also need to add a check here that there are no other deployments in the stateroot before deleting it

Reply:
@donpenney sorry I don't understand the second part, if we want to remove `rhcos_4.13.9` there should be only on directory in `/sysroot/ostree/deploy/`?